test/functional/wallet_basic.py

```diff
@@ -428,6 +429,15 @@ def run_test(self):
         assert_equal(coinbase_tx_1["transactions"][0]["blockhash"], blocks[1])
         assert_equal(len(self.nodes[0].listsinceblock(blocks[1])["transactions"]), 0)
 
+        # Check that we can retrieve the witness nonce from the coinbase
+        coinbase_txid = self.nodes[0].getblock(block_hash)['tx'][0]
+        coinbase_tx_2 = self.nodes[1].gettransaction(txid=coinbase_txid, verbose=True)
+        assert "txinwitness" in coinbase_tx_2["decoded"]["vin"][0]
+        witnesses = coinbase_tx_2["decoded"]["vin"][0]["txinwitness"]
+        assert_equal(len(witnesses), 1)
+        assert_is_hex_string(witnesses[0])
+        assert_equal(len(witnesses[0]), 64)
```

Comment:
```suggestion
        assert_equal(witnesses[0], '00'*32)
```

I think Bitcoin Core uses all zeros as the nonce by default, so you could just directly check this here. Though, your check also works.

Reply:
Yep, it does, and I think I'll switch to your explicit check -- if Bitcoin Core ever changes to some other strategy then having to update the tests as an explicit acknowledgement of that would probably be a good idea. Thanks.